catroid/src/androidTest/java/org/catrobat/catroid/test/io/DefaultDroneProgramLoadingTest.java

```diff
@@ -0,0 +1,113 @@
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.test.io;
+
+import android.content.Context;
+import android.support.test.InstrumentationRegistry;
+import android.support.test.runner.AndroidJUnit4;
+
+import org.catrobat.catroid.ProjectManager;
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.common.defaultprojectcreators.ArDroneProjectCreator;
+import org.catrobat.catroid.content.Project;
+import org.catrobat.catroid.content.Scene;
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.exceptions.LoadingProjectException;
+import org.catrobat.catroid.io.XstreamSerializer;
+import org.catrobat.catroid.test.utils.TestUtils;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+import static junit.framework.Assert.assertEquals;
+
+@RunWith(AndroidJUnit4.class)
+public class DefaultDroneProgramLoadingTest {
+	private final XstreamSerializer storageHandler;
+	private String projectName;
+	private Project currentProjectBuffer;
+	private Context targetContext;
+	private Project project;
+
+	public DefaultDroneProgramLoadingTest() {
+		storageHandler = XstreamSerializer.getInstance();
+	}
+
+	@Before
+	public void setUp() throws Exception {
+		targetContext = InstrumentationRegistry.getTargetContext();
+		currentProjectBuffer = ProjectManager.getInstance().getCurrentProject();
+		projectName = targetContext.getString(R.string.default_drone_project_name);
+		project = new ArDroneProjectCreator().createDefaultProject(projectName, targetContext, false);
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		ProjectManager.getInstance().setProject(currentProjectBuffer);
+		TestUtils.deleteProjects(projectName);
+	}
+
+	@Test
+	public void testDroneProgramLoadingSuccessfully() throws IOException, LoadingProjectException {
+		Project loadedProject = storageHandler.loadProject(projectName, targetContext);
+		Scene preScene = project.getDefaultScene();
+		Scene postScene = loadedProject.getDefaultScene();
+
+		ArrayList<Sprite> preSpriteList = (ArrayList<Sprite>) project.getDefaultScene().getSpriteList();
+		ArrayList<Sprite> postSpriteList = (ArrayList<Sprite>) loadedProject.getDefaultScene().getSpriteList();
+
+		assertEquals(project.getName(), loadedProject.getName());
+		assertEquals(preScene.getName(), postScene.getName());
+
+		assertEquals(preSpriteList.get(0).getName(), postSpriteList.get(0).getName());
```

Comment:
this whole assertEquals block can be rewritten as 
`Assert.assertEquals(preSpriteList, postSpriteList);`

Reply:
ok. done